primitives/src/consensus_types.rs

```diff
@@ -0,0 +1,246 @@
+use codec::{Decode, Encode};
+use ethereum_consensus::primitives::Epoch;
+use ssz_rs::{Deserialize, Node, Vector, List};
+use ssz_rs::prelude::*;
+
+const BLS_PUBLIC_KEY_BYTES_LEN: usize = 48;
+const BLS_SECRET_KEY_BYTES_LEN: usize = 32;
+const BLS_SIGNATURE_BYTES_LEN: usize = 96;
+
+#[derive(Default, Debug, SimpleSerialize, Clone, PartialEq, Eq, codec::Encode, codec::Decode)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct BeaconBlockHeader {
+    #[cfg_attr(feature = "serde", serde(with = "crate::serde::as_string"))]
+    pub slot: u64,
+    #[cfg_attr(feature = "serde", serde(with = "crate::serde::as_string"))]
+    pub proposer_index: u64,
+    pub parent_root: Node,
+    pub state_root: Node,
+    pub body_root: Node,
+}
+
+#[derive(Default, Clone, Debug, SimpleSerialize, PartialEq, Eq, codec::Encode, codec::Decode, )]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Checkpoint {
+    #[cfg_attr(feature = "serde", serde(with = "crate::serde::as_string"))]
+    pub epoch: u64,
+    pub root: Node,
+}
+
+#[derive(Default, Debug, SimpleSerialize, Clone, PartialEq, Eq, codec::Encode, codec::Decode )]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Eth1Data {
+    pub deposit_root: Node,
+    #[cfg_attr(feature = "serde", serde(with = "crate::serde::as_string"))]
+    pub deposit_count: u64,
+    pub block_hash: [u8; 32],
```

Comment:
Let's use Node for all 32 bytes types
```suggestion
    pub block_hash: Node,
```